src/controller/CHIPDeviceControllerFactory.cpp

```diff
@@ -147,7 +147,8 @@ CHIP_ERROR DeviceControllerFactory::InitSystemState(FactoryInitParams params)
 #endif
                                                             ));
 
-    stateParams.fabricTable                                   = chip::Platform::New<FabricTable>();
+    stateParams.fabricTable = (params.fabricTable == nullptr) ? chip::Platform::New<FabricTable>() : params.fabricTable;
```

Comment:
This is not OK, for a few reasons:

1) This will end up calling `Init` on the table below, but in this case it's already initialized, no?
2) `DeviceControllerSystemState::Shutdown` will `Platform::Delete` the fabric table, but it's not allocated via `Platform::*` APIs, possibly leading to memory corruption.  And definitely calling the object's destructor while `Server` is still referencing it... 

There are pre-existing issues here about a second session manager and second InteractionModelEngine Init, by the way, which make me wonder how any of this works at all.

Reply:
ok, I found a different approach and so removed this change, thanks